### ISSUE TITLE
AbstractMeta.getForeignKeys() and other key / index producing methods don't handle namespacing correctly

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11292]
In a schema where multiple tables contain a constraint by the same name, `AbstractMeta` based implementations will produce the wrong lookups.

For example, when doing this:

```java
ctx.execute("drop table if exists t2");
ctx.execute("drop table if exists t1");
ctx.execute("create table t1 (i int constraint uk unique)");
ctx.execute("create table t2 (j int constraint uk unique)");
System.out.println(ctx.meta().getUniqueKeys());
```

Then, only one unique constraint can be retrieved. The meta data is present on the tables, it's just not aggregated correctly. E.g. this works:

```java
System.out.println(ctx.meta().getTables().stream().flatMap(t -> t.getUniqueKeys().stream()).collect(toList()));
```

**Comment by lukaseder** [comment]
The problem is that we're never initialising constraints by their qualified name - neither in generated code, nor in runtime `Meta` objects. We probably should...

**Comment by lukaseder** [comment]
A special case are unnamed constraints, which currently have the special name `""`.  Qualifying `""` with `"T"` leads to `"T"`, not `"T".""`, see https://github.com/jOOQ/jOOQ/issues/9324 and https://github.com/jOOQ/jOOQ/issues/7835.

There are numerous issues dealing with unnamed constraints (e.g. https://github.com/jOOQ/jOOQ/issues/9820, https://github.com/jOOQ/jOOQ/issues/10163). Ultimately, unnamed constraints don't exist. All RDBMS produce some sort of system-generated name. Perhaps we should have an internal way of representing unnamed names, and implement a strategy to generate names for them.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.6 (#11293)

**Comment by lukaseder** [comment]
Separate issue for unnamed constraints: https://github.com/jOOQ/jOOQ/issues/11294
